### ISSUE TITLE
Server side data source with row grouping AND groupIncludeFooter produces max call stack error and grid will not update

### ISSUE THREAD

**Issue by evkw** [ag-grid/ag-grid#6954]
**I'm submitting a ...** 
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**
Please see https://plnkr.co/edit/YGKJOaDsp9YzquCw

In the provided plunker, please open dev console and press the button above the rendered AgGrid to see the error

**Current behavior**
Using server side rendering in conjunction with row grouping AND 'groupIncludeFooter' set to true will create a max call stack error when the data source is changed.

When the 'groupIncludeFooter' is set to true, the rowNode adds itself as a sibling and the destroyRowNode will recursively attempt to delete all rowNodes and its sibilings when the data source is updated. Because each node has itself as sibiling, this recursive function would operate infinitely and hits the max call stack size exceeded error.

![image](https://github.com/ag-grid/ag-grid/assets/15670238/d424a059-6e9d-4a16-903a-fd358974b1ba)

![image](https://github.com/ag-grid/ag-grid/assets/15670238/9c3bf3bd-de9c-43b0-a9db-85b1744449ba)

**Expected behavior**
Updating the data source should not result in an error (this was the case in AgGrid version 27.3.0)
I don't know if having a node be its own sibling is expected or not.

**Please tell us about your environment:**
* Windows 11
* AG Grid version: 30.0.6

**Browser:**
* 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/115.0.0.0 Safari/537.36'

**Comment by evkw** [comment]
as a work around you are able to do something like this before updating the data source. But I don't know what unintended consequences that would have

`
     this.gridApi.forEachNode(node => {
      node.sibling = null;
    })
`

**Comment by zoheil-ag** [comment]
Hello @evkw,

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference: **AG-9183**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

See whether this item will be in the next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Thanks again for bringing this up with us.

Kind regards,
Zoheil

**Comment by isaiahshiner** [comment]
I realize this issue is closed, but this might be the easiest place to communicate that TylerTech/Socrata is also blocked by this issue.

**Comment by AG-Zoheil** [comment]
Thank you for your comment @isaiahshiner.

Please note that a fix for this has been released with our latest version **30.2.0** today.

Kind regards,
Zoheil
